core/src/main/java/org/apache/spark/SparkExecutorInfo.java

```diff
@@ -30,4 +30,8 @@
   int port();
   long cacheSize();
   int numRunningTasks();
+  long usedOnHeapStorageMemory();
```

Comment:
Maybe I'm missing something here, but do we already have a real use case for the added memory metrics here?

Reply:
This information was exposed by the public method being removed by this PR, so it makes sense to add these so that people still have a way to get that data.

Reply:
If you are referring to `StorageStatus`.`onHeapMemUsed`/`offHeapMemUsed`/`onHeapMemRemaining`/`offHeapMemRemaining`, it makes great sense then. But I still don't quite get a few things:
1. Why do we use a different name format, while we could have let them be the same as in `StorageStatus`?
2. Are these information all we need to add to `SparkExecutorInfo` ? After this change, how do we expose the information of disk usage?

Reply:
1. That follows the names in the public `MemoryMetrics` class from the REST API.
2. We could add that, just as we could add a whole lot of other things. At some point we should look at exposing the REST API types directly through `SparkStatusTracker` instead of having these mirror types.

Reply:
I'm not disagreeing with you on making such changes, but I'm also worrying about users could have to change their code a lot because of the changes we made. If you don't mind, may I submit a follow up PR to minimize the gap between the `SparkExecutorInfo` and `StorageStatus`?

Reply:
Sure. But users put themselves at those kind of risks by using @DeveloperApi methods, especially ones that have been deprecated.

Reply:
Agreed, the changes should be a minor issue. Thanks for explanations!